thunder/torch/__init__.py

```diff
@@ -1983,7 +1983,8 @@ def copysign_(a, b, /):
 
 @torchsymbol(torch.Tensor.copy_, is_method=True)  # , tags=(prims.OpTags.IN_PLACE,))
 def copy_(a, b, /):
-    return prims.copy_(b, a)
+    cd = get_compile_data()
+    return prims.copy_(b, a, grad_enabled=cd.is_grad_enabled if cd is not None else False)
```

Comment:
`if cd is None` (probably happens for `thunder.trace` with default arguments), should we assume that we are running with `grad_enabled`? I think that it is likely case. Wdyt?

cc: @IvanYashchuk

Reply:
I'm don't have the background to have an opinion on this.  I defer to you @kshitij12345 and @IvanYashchuk.

Reply:
`cd` is our controlled way of specifying and querying the state of PyTorch. If it's None, I don't think we should do anything special. It's the responsibility of the outside system to set up a correct `cd` object. `is_grad_enabled` is a sensible default because if nothing else is specified, we should assume we are executing a program as given in the "inference" mode with no additional side transformations.